File: 5Programs/C#/WSMT-5pb/WSMT-5pb/Node.cs
namespace WSMT_5pb
{
    public class Node
    {
        public int NodeId { get; set; }
        public int ParentId { get; set; }
        public string Side { get; set; }
        public string Data { get; set; }

        public Node(int nodeId, int parentId, string side, string data)
        {
            NodeId = nodeId;
            ParentId = parentId;
            Side = side;
            Data = data;
        }

        public CustomNode ToCustomNode()
        {
            return new CustomNode(NodeId, null, null, Data);
        }

        public override string ToString()
        {
            return "[NodeId: " + NodeId + " ParentId: " + ParentId + " Side: " + Side + " " + Data + "]";
        }

    }
}

File: 5Programs/NodeJs/routes/node.js
var CustomNode = require("./custom_node.js");
var method = Node.prototype;

function Node(node_id, parent_id, side, data) {
    this.node_id = node_id;
    this.parent_id = parent_id;
    this.side = side;
    this.data = data;
}

method.getNodeId = function () {
    return this.node_id;
};
method.getParentId = function () {
    return this.parent_id;
};
method.getSide = function () {
    return this.side;
};
method.getData = function () {
    return this.data;
};
method.to_custom_node = function () {
    var node = new CustomNode(this.node_id, null, null, this.data);
    return node
};
method.toString = function () {
    return 'Node_id ' + this.node_id + ' Parent_id ' + this.parent_id;
};


module.exports = Node;
